radio_beam/tests/test_beam.py

```diff
@@ -83,36 +95,162 @@ def test_from_casa_image():
-#     beam_2 = radio_beam.Beam(5.*u.arcsec, 3.*u.arcsec, 120.*u.deg)
-
-#     beam_3 = beam_1.convolve(beam_2)
-#     print "test1: ",beam_2 == beam_3.deconvolve(beam_1)
-#     print "test1: ",beam_2 - beam_3.deconvolve(beam_1)
-#     print "test2: ",beam_1 == beam_3.deconvolve(beam_2)
-#     print "test2: ",beam_1 - beam_3.deconvolve(beam_2)
-#     print "beam1: ",beam_1
-#     print "beam2: ",beam_2
-#     print "beam3: ",beam_3
-#     print "beam3.deconv(beam2): ",beam_3.deconvolve(beam_2)
-#     np.testing.assert_almost_equal(beam_3.deconvolve(beam_2).sr.value, beam_1.sr.value)
-
-#     # Area
-#     print beam_3.sr
-#     print beam_2.sr
-#     #  <Quantity 3.994895404940742e-10 sr>
-#     print beam_1.sr
-#     print beam_1
-
-#     # Janskies to Kelvin
-#     np.testing.assert_almost_equal(beam_2.jtok(1.e9), 81474.701386)
-    # 81474
-
-    # Return as array
-
-    # Return as array given WCS
+    bima_casa_beam = Beam.from_casa_image(fname)
+
+
+def test_attach_to_header():
+    fname = data_path("NGC0925.bima.mmom0.fits.gz")
+    hdr = fits.getheader(fname)
+    hdr_copy = hdr.copy()
+    del hdr_copy["BMAJ"], hdr_copy["BMIN"], hdr_copy["BPA"]
+
+    bima_beam = Beam.from_fits_header(fname)
+
+    new_hdr = bima_beam.attach_to_header(hdr_copy)
+
+    npt.assert_equal(new_hdr["BMAJ"], hdr["BMAJ"])
+    npt.assert_equal(new_hdr["BMIN"], hdr["BMIN"])
+    npt.assert_equal(new_hdr["BPA"], hdr["BPA"])
+
+
+def test_beam_projected_area():
+
+    distance = 250 * u.pc
+
+    major = 0.1 * u.rad
+    beam = Beam(major, major, 30 * u.deg)
+
+    beam_sr = (major**2 * 2 * np.pi / (8 * np.log(2))).to(u.sr)
+
+    assert_quantity_allclose(beam_sr.value * distance ** 2,
+                             beam.beam_projected_area(distance))
+
+
+def test_jtok():
+
+    major = 0.1 * u.rad
+    beam = Beam(major, major, 30 * u.deg)
+
+    freq = 1.42 * u.GHz
+
+    conv_factor = u.brightness_temperature(beam.sr, freq)
+
+    assert_quantity_allclose((1 * u.Jy).to(u.K, equivalencies=conv_factor),
+                             beam.jtok(freq))
+
+
+def test_jtok_equiv():
+
+    major = 0.1 * u.rad
+    beam = Beam(major, major, 30 * u.deg)
+
+    freq = 1.42 * u.GHz
+
+    conv_factor = u.brightness_temperature(beam.sr, freq)
+    conv_beam_factor = beam.jtok_equiv(freq)
+
+    assert_quantity_allclose((1 * u.Jy).to(u.K, equivalencies=conv_factor),
+                             (1 * u.Jy).to(u.K, equivalencies=conv_beam_factor))
+
+    assert_quantity_allclose((1 * u.K).to(u.Jy, equivalencies=conv_factor),
+                             (1 * u.K).to(u.Jy, equivalencies=conv_beam_factor))
+
+
+def test_convolution():
+
+    major1 = 1 * u.deg
+    minor1 = 0.5 * u.deg
+    pa1 = 0.0 * u.deg
+    beam1 = Beam(major1, minor1, pa1)
+
+    major2 = 1 * u.deg
+    minor2 = 0.75 * u.deg
+    pa2 = 90.0 * u.deg
+    beam2 = Beam(major2, minor2, pa2)
+
+    alpha = (major1 * np.cos(pa1))**2 + (minor1 * np.sin(pa1))**2 + \
```

Comment:
Unless you re-derived this from scratch, which would be awesome, include a note above that this math was originally done in `https://github.com/pkgw/carma-miriad/blob/CVSHEAD/src/subs/gaupar.for`
 by [Bob Sault](http://www.ph.unimelb.edu.au/~rsault/)  (so we have some idea of where the math is coming from)